### PR TITLE
config: remove signature_key_algorithm

### PR DESCRIPTION
**Pull request by calebdoxsey**
## Summary
Remove `signing_key_algorithm` and infer it from `signing_key`.

## Related issues
Fixes https://github.com/pomerium/internal/issues/521

## Checklist
- [x] reference any related issues
- [x] updated docs
- [x] updated unit tests
- [x] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

File: pkg/cryptutil/jose.go [hunk 1]
```diff
@@ -93,3 +97,15 @@ func loadPublicKey(b []byte) (interface{}, error) {
 
 	return nil, fmt.Errorf("couldn't load public key: %w", wrappedErr)
 }
+
+// SignatureAlgorithmForKey returns the signature algorithm for the given key.
+func SignatureAlgorithmForKey(key interface{}) (jose.SignatureAlgorithm, error) {
```
Comment (desimone) [inline pkg/cryptutil/jose.go:102]: Could we add a few unit tests that explicitly test this function for both private and public keys for both algo types?

**Review by desimone — CHANGES_REQUESTED** [review]
LGTM. Only suggestion would be to directly test the exported `SignatureAlgorithmForKey` function.

File: docs/docs/upgrading.md [hunk 2]
```diff
@@ -10,6 +10,8 @@ description: >-
 ### Removed options
 The deprecated `headers` option has been removed. Use [`set_response_headers`](/reference/readme.md#set-response-headers) instead.
 
+The `signing_key_algorithm` option has been removed and will now be inferred from `signing_key`.
```
Comment (alexfornuto) [inline docs/docs/upgrading.md:13]: Is this being backported to 15.0?

Comment (calebdoxsey) [reply docs/docs/upgrading.md:13]: No it should not be backported.

**Review by alexfornuto — APPROVED** [review]

**Review by desimone — APPROVED** [review]

**Pull request merged**